gluten-core/src/main/java/org/apache/gluten/vectorized/JniLibLoader.java

```diff
@@ -125,13 +139,10 @@ public static synchronized void unloadFromPath(String libPath) {
       return;
     }
 
+    LOG.info("Starting unload library path: {} ", libPath);
     REQUIRE_UNLOAD_LIBRARY_PATHS.remove(libPath);
 
     try {
-      while (Files.isSymbolicLink(Paths.get(libPath))) {
```

Comment:
When running on yarn cluster, `libPath` is a symbolic link existed on the working directory of yarn container, like `/data5/hadoop/yarn/local/usercache/xumingyong/appcache/application_1710462985812_777680/container_e57_1710462985812_777680_01_000321/./libch.so`. It's an `APPLICATION` level resource and will be removed when container exists. So it may have been removed here when reading real path, which causes `Files.isSymbolicLink(Paths.get(libPath))` return false in that the library cannot be unload. As a workaround, we can read real path when loading the library.